bin/improver-threshold

```diff
@@ -131,12 +164,38 @@ def main():
         thresholds = args.threshold_values
         fuzzy_bounds = None
 
-    result = BasicThreshold(
+    result_no_realization = BasicThreshold(
         thresholds, fuzzy_factor=args.fuzzy_factor,
         fuzzy_bounds=fuzzy_bounds, threshold_units=args.threshold_units,
         below_thresh_ok=args.below_threshold).process(cube)
 
-    save_netcdf(result, args.output_filepath)
+    if args.collapse_coord == "realization":
```

Comment:
The only part of this code block that is specific to `realization` is the choice of slope and y0val, and then that is only specific to anything that wants to have equal weighting. It might be better to make this if statement just `if args.collapse_coord` and then you could support collapsing across any coordinate, which would make this a bit more generic. Although I notice that that as the default argument for the `collapse_coord` argument is `"None"`, rather than `None`, you might want to do either `if args.collapse_coord == "None"` or convert "None" to None e.g. as in https://stackoverflow.com/questions/26481774/is-there-a-simple-way-to-convert-the-string-none-to-a-proper-none-variable-in?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa.

We might want to consider whether we want to support options for the `weighted_mode` of `weighted_mean` and `weighted_maximum`, and whether we want to support the `wts_calc_method` option to allow linear or non-linear weights to be specified.

Reply:
I'd suggest we just go for linear, no slope (equal weights across the given coordinate). If people want the full functionality they can always call weighted blend separately